packages/rum/src/viewCollection.ts

```diff
@@ -39,35 +39,38 @@ export const SESSION_KEEP_ALIVE_INTERVAL = 5 * ONE_MINUTE
 export function startViewCollection(location: Location, lifeCycle: LifeCycle) {
   let currentLocation = { ...location }
   const startOrigin = 0
-  let currentView = newView(lifeCycle, currentLocation, ViewLoadingType.INITIAL_LOAD, startOrigin)
+  let currentView = newView(lifeCycle, ViewLoadingType.INITIAL_LOAD, startOrigin)
+  currentView.triggerUpdate(currentLocation)
 
   // Renew view on history changes
   trackHistory(() => {
     if (areDifferentViews(currentLocation, location)) {
-      currentLocation = { ...location }
-      currentView.triggerUpdate()
+      currentView.triggerUpdate(currentLocation)
```

Comment:
Two things are bothering me:
- I find it a bit confusing to update location of the view from a view update
- It is a bit weird to maintain a current location on startViewCollection scope and a location on newView scope

What about:
- maintaining the view location only in the view scope, areDifferentViews could be moved in current view
- have a triggerUpdate and an updateLocation in order to separate the two concerns

wdyt?

Reply:
If `areDifferentViews` usage is done in the `newView` function, it would mean that the `newView` is responsible for ending itself, so we'd need another callback to create a new view after that. Does it seem ok to you?

Reply:
I was thinking about declaring `areDifferentViews` inside new view but using it outside, just for the view location access

Reply:
So you'd have something like:
```ts
trackHistory(() => {
  if (currentView.isDifferentView(newLocation)) {
    currentView.triggerUpdate()
    currentView.end()
    currentView = newView(...)
} else {
   currentView.updateLocation(newLocation)
}
```
right?

Reply:
yes